.github/workflows/tests.yml

```
@@ -12,12 +12,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        branch: ["3.14", "3.13", "3.12"]
+        python-version: ["3.15", "3.14", "3.13", "3.12"]
+        include:
+          - python-version: "3.15"
+            branch: "main"
```

A possible improvement is to compute the branch once and assign it to an env var:
```suggestion
        include:
          - python-version: "3.15"
            branch: "main"
      env:
        # set the branch name to either 'main' or the version number
        BRANCH: ${{ matrix.branch || matrix.version }}
```
This keeps the logic closer and avoids duplication, but since we only need the branch once in the workflow, it might be overkill.